tests/scripts/all.sh

```diff
@@ -1290,6 +1290,38 @@ component_test_no_use_psa_crypto_full_cmake_asan() {
     if_build_succeeded env OPENSSL_CMD="$OPENSSL_NEXT" tests/compat.sh -e '^$' -f 'ARIA\|CHACHA'
 }
 
+component_test_psa_crypto_config_basic() {
+    # full plus MBEDTLS_PSA_CRYPTO_CONFIG
+    msg "build: full + MBEDTLS_PSA_CRYPTO_CONFIG"
+    scripts/config.py full
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
+    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
+    # Need to define the correct symbol and include the test driver header path in order to build with the test driver
+    make CC=gcc CFLAGS="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST -I../tests/include -O2" LDFLAGS="$ASAN_CFLAGS"
+
+    msg "test: psa crypto config basic"
+    make test
+}
+
+component_test_psa_want_ecdsa_disabled_software() {
+    # full plus MBEDTLS_PSA_CRYPTO_CONFIG with PSA_WANT_ALG_ECDSA
+    # without MBEDTLS_ECDSA_C
+    msg "build: full + MBEDTLS_PSA_CRYPTO_CONFIG + PSA_WANT_ALG_ECDSA without MBEDTLS_ECDSA_C"
+    scripts/config.py full
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_DRIVERS
+    scripts/config.py unset MBEDTLS_USE_PSA_CRYPTO
+    scripts/config.py unset MBEDTLS_ECDSA_C
+    scripts/config.py -f include/psa/crypto_config.h set PSA_WANT_ALG_ECDSA
```

Comment:
Just to be clear, are we wanting to specifically handle dependencies in this file when setting the MBEDTLS_xxx features? Seems like there is already a mechanism in place in config.h and duplicating that in some form in this file may not be the right choice. If we do want to do that in this file, I will confirm those options here before making the correction to avoid more churn on this PR. For the second comment in all.sh I will remove the redundant setting.

Reply:
@ronald-cron-arm Looks like you were correct on the dependencies for MBEDTLS_ECDSA_DETERMINISTIC, it does need the MBEDTLS_MD_C and MBEDTLS_HMAC_DRBG_C. @gilles-peskine-arm Do you know if there is s missing dependency on ECDSA_C for deterministic?

Reply:
There's no dependency “handling” in `mbedtls/config.h`, only documentation. There's `mbedtls/check_config.h`, but all it does is to reject certain combinations, it doesn't automatically enable dependencies.

`MBEDTLS_ECDSA_DETERMINISTIC` only has an effect when `MBEDTLS_ECDSA_C` is enabled. I don't know what the right thing is with respect to such cases (as opposed to dependencies where if you turn on A without B, the library doesn't work). Looking at existing cases, we don't seem to be consistent: `MBEDTLS_ECDSA_DETERMINISTIC` doesn't document a dependency on `MBEDTLS_ECDSA_C`, and `MBEDTLS_ECP_RESTARTABLE` doesn't document a dependency on `MBEDTLS_ECP_C`, but `MBEDTLS_PKCS1_V15` and `MBEDTLS_PKCS1_V21` document a dependency on `MBEDTLS_RSA_C`.

Going back to the original question, `set PSA_WANT_ALG_ECDSA` is redundant since it's on by default, but making it explicit is arguably good documentation. So I'm fine with either keeping it in or leaving it out.

Reply:
@ronald-cron-arm Sounds like since there is not a hard set requirement to perform the dependencies here, we should leave it for a later ticket to determine if we want to make those types of changes. Would it be best just to keep this as is for the first round of implementation of these features, then work on determining the best path for dependency setup/check based on the feature. I will also just leave in the current sets for set PSA_WANT_ALG_ECDSA per @gilles-peskine-arm comment so we can move this PR along.

Reply:
Fine by me to keep "scripts/config.py -f include/psa/crypto_config.h set PSA_WANT_ALG_ECDSA" lines.